geode-core/src/test/java/org/apache/geode/internal/cache/tx/RemotePutMessageTest.java

```diff
@@ -22,44 +23,43 @@
 import java.util.Set;
 
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.powermock.api.mockito.PowerMockito;
-import org.powermock.core.classloader.annotations.PowerMockIgnore;
-import org.powermock.core.classloader.annotations.PrepareForTest;
-import org.powermock.modules.junit4.PowerMockRunner;
 
+import org.apache.geode.CancelCriterion;
 import org.apache.geode.cache.RegionDestroyedException;
+import org.apache.geode.distributed.internal.DistributionManager;
+import org.apache.geode.distributed.internal.InternalDistributedSystem;
 import org.apache.geode.distributed.internal.membership.InternalDistributedMember;
 import org.apache.geode.internal.cache.CacheDistributionAdvisor;
 import org.apache.geode.internal.cache.DistributedRegion;
 import org.apache.geode.internal.cache.EntryEventImpl;
+import org.apache.geode.internal.cache.EventID;
 import org.apache.geode.internal.cache.RemoteOperationException;
 
-@PowerMockIgnore("*.UnitTest")
-@RunWith(PowerMockRunner.class)
-@PrepareForTest(RemotePutMessage.class)
 public class RemotePutMessageTest {
   @Test
   public void testDistributeNotFailWithRegionDestroyedException() throws RemoteOperationException {
     EntryEventImpl event = mock(EntryEventImpl.class);
+    EventID eventID = mock(EventID.class);
     DistributedRegion region = mock(DistributedRegion.class);
+    InternalDistributedSystem ids = mock(InternalDistributedSystem.class);
+    DistributionManager dm = mock(DistributionManager.class);
+    CancelCriterion cc = mock(CancelCriterion.class);
     CacheDistributionAdvisor advisor = mock(CacheDistributionAdvisor.class);
     InternalDistributedMember member = mock(InternalDistributedMember.class);
     Set<InternalDistributedMember> replicates = new HashSet<>(Arrays.asList(member));
     RemotePutMessage.RemotePutResponse response = mock(RemotePutMessage.RemotePutResponse.class);
     Object expectedOldValue = new Object();
 
     when(event.getRegion()).thenReturn(region);
+    when(event.getEventId()).thenReturn(eventID);
     when(region.getCacheDistributionAdvisor()).thenReturn(advisor);
     when(advisor.adviseInitializedReplicates()).thenReturn(replicates);
     when(response.waitForResult()).thenThrow(new RegionDestroyedException("", ""));
-
-    PowerMockito.mockStatic(RemotePutMessage.class);
-    PowerMockito
-        .when(RemotePutMessage.distribute(event, 1, false, false, expectedOldValue, false, false))
-        .thenCallRealMethod();
-    PowerMockito.when(RemotePutMessage.send(member, region, event, 1, false, false,
-        expectedOldValue, false, false, false)).thenReturn(response);
+    when(region.getSystem()).thenReturn(ids);
+    when(region.getDistributionManager()).thenReturn(dm);
+    when(ids.getDistributionManager()).thenReturn(dm);
+    when(dm.getCancelCriterion()).thenReturn(cc);
+    when(dm.putOutgoing(any())).thenReturn(null);
 
     RemotePutMessage.distribute(event, 1, false, false, expectedOldValue, false, false);
```

Comment:
This isn't really related to your PR, but this test seems odd to me.  It doesn't do any validations, though I guess the validation is that the test doesn't throw RegionDestroyedException?  Seems like this test could be improved, but again, I'd say it isn't related to PowerMock.